internal/impl/pure/scanner_csv.go

```diff
@@ -5,15 +5,18 @@ import (
 	"encoding/csv"
 	"errors"
 	"io"
+	"slices"
 
 	"github.com/warpstreamlabs/bento/public/service"
 )
 
 const (
-	scsvFieldCustomDelimiter = "custom_delimiter"
-	scsvFieldParseHeaderRow  = "parse_header_row"
-	scsvFieldLazyQuotes      = "lazy_quotes"
-	scsvFieldContinueOnError = "continue_on_error"
+	scsvFieldCustomDelimiter        = "custom_delimiter"
+	scsvFieldParseHeaderRow         = "parse_header_row"
+	scsvFieldLazyQuotes             = "lazy_quotes"
+	scsvFieldContinueOnError        = "continue_on_error"
+	scsvFieldExpectedHeaders        = "expected_headers"
+	scsvFieldExpectedNumberOfFields = "expected_number_of_fields"
```

Comment:
Why not:

```suggestion
	scsvFieldExpectedNumberOfFields = "expected_number_of_headers"
```

IMO using different terminology (`_headers` versus `_fields`) is a bit confusing. Wdyt?

Reply:
Yes I thought this - but there is a slight difference - headers is referring to the first row in a csv file that would typically contain column names - with `expected_headers` we are checking the value of the column names. 

But with `expected_number_of_fields` where 'field' would refer to a value in a row, we are checking that there is the correct amount of values in a row.

So changing the 'expected_number_of_fields' to 'expected_number_of_headers' wouldn't make sense because it would still work if `parse_header_row` was set to false indicating that there aren't headers in the file. 

We could change `expected_headers` to `expected_field_names` perhaps but the other already existing fields do use the terminology 'headers' so I think it's ok as is.